spring-batch-infrastructure/src/main/java/org/springframework/batch/item/data/builder/Neo4jItemReaderBuilder.java

```diff
@@ -0,0 +1,210 @@
+	 * @return this instance for method chaining
+	 * @see Neo4jItemReader#setReturnStatement(String)
+	 */
+	public Neo4jItemReaderBuilder<T> returnStatement(String returnStatement) {
+		this.returnStatement = returnStatement;
+
+		return this;
+	}
+
+	/**
+	 * An optional match fragment of the cypher query. MATCH is prepended to the statement
+	 * provided and should <em>not</em> be included.
+	 *
+	 * @param matchStatement the match fragment of the cypher query
+	 * @return this instance for method chaining
+	 * @see Neo4jItemReader#setMatchStatement(String)
+	 */
+	public Neo4jItemReaderBuilder<T> matchStatement(String matchStatement) {
+		this.matchStatement = matchStatement;
+
+		return this;
+	}
+
+	/**
+	 * An optional where fragment of the cypher query. WHERE is prepended to the statement
+	 * provided and should <em>not</em> be included.
+	 *
+	 * @param whereStatement where fragment of the cypher query
+	 * @return this instance for method chaining
+	 * @see Neo4jItemReader#setWhereStatement(String)
+	 */
+	public Neo4jItemReaderBuilder<T> whereStatement(String whereStatement) {
+		this.whereStatement = whereStatement;
+
+		return this;
+	}
+
+	/**
+	 * A list of properties to order the results by. This is required so that subsequent
+	 * page requests pull back the segment of results correctly. ORDER BY is prepended to
+	 * the statement provided and should <em>not</em> be included.
+	 *
+	 * @param orderByStatement order by fragment of the cypher query.
+	 * @return this instance for method chaining
+	 * @see Neo4jItemReader#setOrderByStatement(String)
+	 */
+	public Neo4jItemReaderBuilder<T> orderByStatement(String orderByStatement) {
+		this.orderByStatement = orderByStatement;
+
+		return this;
+	}
+
+	/**
+	 * The object type to be returned from each call to {@link Neo4jItemReader#read()}
+	 *
+	 * @param targetType the type of object to return.
+	 * @return this instance for method chaining
+	 * @see Neo4jItemReader#setTargetType(Class)
+	 */
+	public Neo4jItemReaderBuilder<T> targetType(Class<T> targetType) {
+		this.targetType = targetType;
+
+		return this;
+	}
+
+	/**
+	 * Returns a fully constructed {@link Neo4jItemReader}.
+	 *
+	 * @return a new {@link Neo4jItemReader}
+	 */
+	public Neo4jItemReader<T> build() {
+		if (this.saveState) {
+			Assert.hasText(this.name, "A name is required when saveState is set to true");
+		}
+		Assert.notNull(this.sessionFactory, "sessionFactory is required.");
+		Assert.notNull(this.targetType, "targetType is required.");
+		Assert.hasText(this.startStatement, "startStatement is required.");
+		Assert.hasText(this.returnStatement, "returnStatement is required.");
+		Assert.hasText(this.orderByStatement, "orderByStatement is required.");
+
+		Neo4jItemReader<T> reader = new Neo4jItemReader<>();
+		reader.setMatchStatement(this.matchStatement);
+		reader.setOrderByStatement(this.orderByStatement);
+		reader.setPageSize(this.pageSize);
```

Comment:
Simple enough.  Do we want to check the pageSize in the setPageSize method (maxItemCount for that matter as well) in the AbstractPaginatedDataItemReader, so that all the readers can take advantage.